Source/MLXNN/Activations.swift

```diff
@@ -48,7 +48,8 @@ public func relu(_ x: MLXArray) -> MLXArray {
 /// - <doc:activations>
 /// - ``LeakyReLU``
 public func leakyRelu(_ x: MLXArray, negativeSlope: Float = 0.01) -> MLXArray {
-    maximum(negativeSlope * x, x)
+    let negativeSlope = negativeSlope.asMLXArray(dtype: x.dtype)
+    return compiledLeakyRelu(x, negativeSlope)
```

Comment:
Pattern for compiling.  This goes with this change further along in the file:

```
private let compiledLeakyRelu: (MLXArray, MLXArray) -> MLXArray = {
    compile(shapeless: true) { x, negativeSlope in
        maximum(negativeSlope * x, x)
    }
}()
```

Note: I am not compiling code that is just a single function call:

```
public func relu(_ x: MLXArray) -> MLXArray {
    maximum(x, 0)
}
```